src/Mod/PartDesign/Gui/CommandPrimitive.cpp

```diff
@@ -89,8 +89,9 @@ void CmdPrimtiveCompAdditive::activated(int iMsg)
         if ( doc->getObjectsOfType(PartDesign::Body::getClassTypeId()).empty() ) {
             shouldMakeBody = true;
         } else {
-            PartDesignGui::needActiveBodyError();
-            return;
+            pcActiveBody = PartDesignGui::needActiveBodyMessage(doc);
+            if (pcActiveBody == nullptr)
```

Comment:
Minor nitpick -- the Core Guidelines suggest this be written:
```suggestion
            if (!pcActiveBody)
```
(See http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es87-dont-add-redundant--or--to-conditions)

Reply:
I could have sworn I made this change after you suggested this back when I was working on this. Maybe got lost in the rebase.